compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -766,9 +766,11 @@ impl<'a, 'b> Context<'a, 'b> {
         // of each variable because we don't want to move out of the arguments
         // passed to this function.
         for (i, e) in self.args.into_iter().enumerate() {
-            for arg_ty in self.arg_unique_types[i].iter() {
-                args.push(Context::format_arg(self.ecx, self.macsp, e.span, arg_ty, i));
-            }
+            args.extend(
+                self.arg_unique_types[i]
+                    .iter()
+                    .map(|arg_ty| Context::format_arg(self.ecx, self.macsp, e.span, arg_ty, i)),
+            );
             heads.push(self.ecx.expr_addr_of(e.span, e));
         }
         for pos in self.count_args {
```

Comment:
This loop could also be rewritten to `args.extend(self.count_args.map(|| {}))`. In that case the advantage would be that extend would know exactly how much space to reserve instead of having to grow on pushes.